frontend/public/components/utils/firehose.jsx

```diff
@@ -118,15 +118,18 @@ const stateToProps = ({ k8s }, { resources }) => {
     (models, { kind }) => models.set(kind, k8s.getIn(['RESOURCES', 'models', kind])),
     ImmutableMap(),
   );
-  const loaded = (r) =>
-    r.optional ||
-    k8s.getIn([
+  const loaded = (r) => {
+    if (_.isUndefined(k8sModels.get(r.kind))) {
+      return false;
+    }
```

Comment:
I think this could be removed and you'd get the same result since the lines below would evaluate to `undefined`.

Either way, this would be a regression of https://bugzilla.redhat.com/show_bug.cgi?id=1731480 :(